packages/backend/src/utils.ts

```
@@ -27,3 +30,41 @@ export const isDbPinnedChart = (
     data: DbPinnedChart | DbPinnedDashboard,
 ): data is DbPinnedChart =>
     'saved_chart_uuid' in data && !!data.saved_chart_uuid;
+
+export const wrapSentryTransaction = async (
+    name: string,
+    context: CustomSamplingContext,
+    funct: () => Promise<any>,
+) => {
+    const startTime = Date.now();
+    const transaction = Sentry.getCurrentHub()?.getScope()?.getTransaction();
+
+    Logger.debug(
+        `Starting sentry transaction ${
+            transaction?.spanId
+        } "${name}" with context: ${JSON.stringify(context)}`,
+    );
+
+    const span =
+        transaction &&
+        transaction.startChild({
+            op: name,
+            data: context,
+        });
+    try {
+        return await funct();
+    } catch (error) {
+        Logger.error(
+            `Error in wrapped sentry transaction ${transaction?.spanId} "${name}": ${error}`,
+        );
+        Sentry.captureException(error);
+        return null;
```

I would expect to throw the same error instead of null